### DISCUSSION TITLE
I'm confused about security notice

### DISCUSSION THREAD

**Discussion by CycloneRing** [nothings/stb#1584] [Q&A] [answered]
Hi, I can't understand what kind of security issue this library has? How can someone perform attacks when using stb headers when they are compiled in machine code in final program? Can I get some examples?

**Comment by nothings** [comment]
If you try to use stb_image to load an image provided by an attacker, bad things could potentially happen. If you only use it to load your own choice of images, generally things are fine.

**Reply by CycloneRing** [reply]
Thank you for quick reply! Gladly I only need to use it to encode images.

**Reply by CycloneRing** [reply]
Does this apply to PNG or JPG as well or more complicated formats like HDR EXR TGA?
I only need JPG and PNG

**Reply by nothings** [reply, marked as answer]
If you're just using it to encode images it doesn't matter, but it applies to all formats (jpg and png are also complicated internally).
